CMakeLists.txt

```diff
@@ -1,22 +1,65 @@
+cmake_minimum_required(VERSION 3.22)
 PROJECT( twbt )
+set( TWBT_VER "6.xx" )
 
-SET( PROJECT_SRCS twbt.cpp )
-SET( TWBT_VER "5.xx" )
+SET(CMAKE_VERBOSE_MAKEFILE ON)
+if(DEFINED dfhack_SOURCE_DIR)
+    set(DFHACK ${dfhack_SOURCE_DIR})
+elseif(EXISTS ../dfhack) # sitting next to each other top level
+    set(DFHACK ../dfhack)
+elseif(EXISTS ../../dfhack) # twbt is inside a dfhack directory, probably dfhack/plugins
+    set(DFHACK ../../dfhack)
+elseif(EXISTS ../../../dfhack) # twbt is inside a directory in dfhack/plugins (probably)
+    set(DFHACK ../../../dfhack)
+else()
+    message("couldn't find a dfhack directory nearby, modify the cmake and put your dfhack path")
+    SET(DFHACK /your/path/dfhack)
+endif()
 
-STRING( REGEX REPLACE "\\." "" DV_VS "${DF_VERSION}" )
+include(${DFHACK}/CMake/Common.cmake)
+message("Version information\n-------------------")
+message("dwarf fortress: ${DF_VERSION}")
+message("dfhack:         ${DFHACK_VERSION}")
+message("twbt:           ${TWBT_VER}")
+string(REPLACE "." "" DF_VERSION ${DF_VERSION})
 
-SET( CMAKE_CXX_FLAGS "-m64 -O3 -DDFHACK_VERSION=\"${DFHACK_VERSION}\" -DDF_${DV_VS} -DTWBT_VER=\"\\\"${TWBT_VER}\\\"\"" )
+# include directories
+include_directories(${DFHACK}/library/include)
+include_directories(${DFHACK}/library/proto)
+include_directories(${DFHACK}/depends/protobuf)
+include_directories(${DFHACK}/depends/lua/include)
 
-IF( APPLE )
-  IF(CMAKE_COMPILER_IS_GNUCXX)
-    SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++0x -Wno-multichar -framework OpenGL -undefined dynamic_lookup" )
-  ELSE()
-    SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++0x -stdlib=libstdc++ -Wno-tautological-compare -framework OpenGL -mmacosx-version-min=10.6 -undefined dynamic_lookup" )
-  ENDIF()
-ELSE( APPLE )
-  SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x" )
-ENDIF( APPLE )
+# source files
+set( SRC twbt.cpp )
+set( DEP renderer.hpp config.hpp dungeonmode.hpp dwarfmode.hpp renderer_twbt.h commands.hpp plugin.hpp tileupdate_text.hpp tileupdate_map.hpp patches.hpp zoomfix.hpp buildings.hpp items.hpp units.hpp Makefile legacy/renderer_legacy.hpp legacy/twbt_legacy.hpp)
 
-DFHACK_PLUGIN( twbt ${PROJECT_SRCS} )
+find_package(PNG)
+if(NOT PNG_FOUND)
+    # XXX Probably this is wrong. It appears df-twbt/libpng has precompiled
+    # libs for Windows? Probably it should be a submodule that has the
+    # actual libpng source (which builds with CMake).
```

Comment:
My guess is that it was done deliberately because libpng is difficult to build, yet is stable enough that TWBT would almost never need to upgrade it. That's a valid decision to make - DFHack does the same thing for zlib and libruby. I personally wouldn't worry about it.

Reply:
I fully don't remember writing this. And it seems to build without it. maybe it's merged from that cmake_common

Reply:
indeed it is.